src/app/effects/knowledge.ts

```diff
@@ -39,6 +39,8 @@ export class KnowledgeEffects {
       this.knowledgeservice.getSearchResults(querypay.query)
         .takeUntil(nextSearch$)
         .subscribe((response) => {
+          if (response.query) {
+            if (response.query.pages) {
```

Comment:
It is not really required, the above if statement fixes the issue.

Reply:
Since we are assigning value of `response.query.pages` to a variable which will be used later to extract results therefore I am using an extra check to make sure its value is never `undefined` or `null` .